main.c

```diff
@@ -874,34 +874,35 @@ add_solvables_from_file_to_pile (const char  *filename,
 
   if (ret != G_IO_STATUS_EOF)
     {
-      g_prefix_error (error, "Failure while reading file '%s': ", filename);
-      return;
+      g_prefix_error (error, "%s: ", filename);
+      return FALSE;
     }
 
-  /* make sure any error is cleared to indicate success */
-  g_clear_error (error);
+  g_io_channel_unref (ch);
```

Comment:
Move this unref to just before the last `if` otherwise you leak `ch` if `ret != G_IO_STATUS_NORMAL`.